research/04-problem-def.tex

```diff
@@ -1,6 +1,33 @@
 \section{Problem Definition}
 \label{sec:problem-definition}
 
+The previous sections provided the necessary background knowledge of the problem
+domain. Now that the background knowledge has been discussed,
+this section will give the problem definition of the client.
+
+Spoofax offers a wide variety of tools to develop DSLs and accompanying IDE support.
+Precisely because Spoofax is concerned with developing DSLs, rapid prototyping
+of syntax and grammar is a convenient addition to the product.
+Section 2 showed that REPLs provide precisely this rapid prototyping ability
+for other languages like LISP and Python. Therefore, the client has expressed
```

Comment:
`Lisp`. LISP is the historically used, it's 2016 now :P In all seriousness though, section 2 uses Lisp so we should be consistent.